pravega/bookkeeper-operator
# Issue 100: Adding support for test versions

Signed-off-by: SrishT <Srishti.Thakkar@dell.com>

### Change log description
This PR adds support to provide an upgrade path as well as to add an entry corresponding to an unreleased test version, which would enable testing both installation and upgrade of the bookkeeper cluster to non-released test versions.

### Purpose of the change
Fixes #100 

### What the code does
When the user enables `testmode` and provides `version` within the values.yaml
- the version is appended to the upgrade path of each version within the version map, thereby allowing an upgrade from any version to this version.
- a new entry is created in the version map corresponding to this version, which allows this version to be installed.

When additionally the user also provides a value for `fromVersion` in addition to the above values
- a new entry is created to provide an upgrade path from `fromVersion` to `version`

### How to verify it
- Inside the values.yaml, set the value of testmode.enabled to true and provide values for `testmode.version`. This should append `testmode.version` to each of the entries in the bk-supported-versions configmap, and should also create a new entry corresponding to this version
```
<key 1> : <value 1.1> , ... , <version>
<key 2> : <value 2.1> , ... , <version>
....
....
<version>: <version>
```
- Now also provide a value for `testmode.fromVersion` in addition to the above values. This should create an entry of the following format added to the bk-supported-versions configmap
```
<fromVersion> : <fromVersion> , <version>
```
- set the value of testmode.enabled to false. In this case you should not see any changes in the contents of the bk-supported-versions configmap
